tests/plugins/test_deepspeed_plugin.py

```diff
@@ -921,3 +921,47 @@ def test_dataloader(self):
     )
     trainer.fit(model, datamodule=TestSetupIsCalledDataModule())
     trainer.test(model)
+
+
+@mock.patch("torch.optim.lr_scheduler.StepLR.step", autospec=True)
+@RunIf(min_gpus=1, deepspeed=True, special=True)
+def test_deepspeed_scheduler_step_count(mock_step):
+    """Test to ensure that the scheduler is called the correct amount of times during training when scheduler is
+    set to step."""
+    _run_scheduler_test(mock_step, max_epoch=2, limit_train_batches=2, interval="step")
+
+
+@mock.patch("torch.optim.lr_scheduler.StepLR.step", autospec=True)
+@RunIf(min_gpus=1, deepspeed=True, special=True)
+def test_deepspeed_scheduler_step_count_epoch(mock_step):
+    """Test to ensure that the scheduler is called the correct amount of times during training when scheduler is
+    set to epoch."""
+    _run_scheduler_test(mock_step, max_epoch=2, limit_train_batches=2, interval="epoch")
```

Comment:
maybe parametrize the interval here and put the below function logic inside the test?? else looks good!!

Reply:
Oh sorry I should've mentioned, I couldn't make this work (probably because its a special test?) I had to split it up like this unfortunately :(